website/search_migration/migrate.py

```diff
@@ -4,54 +4,107 @@
 from __future__ import absolute_import
 
 import logging
+
+from elasticsearch import helpers
 from modularodm.query.querydialect import DefaultQueryDialect as Q
-from scripts import utils as script_utils
-from website.models import Node
+
+from website import settings
 from framework.auth import User
+from website.models import Node
+from website.app import init_app
 import website.search.search as search
+from scripts import utils as script_utils
+from website.search.elastic_search import es
 
-from website.app import init_app
 
 logger = logging.getLogger(__name__)
 
 app = init_app("website.settings", set_backends=True, routes=True)
+logger = logging.getLogger(__name__)
 
 
-def migrate_nodes():
+def migrate_nodes(index):
+    logger.info("Migrating nodes")
     n_iter = 0
     nodes = Node.find(Q('is_public', 'eq', True) & Q('is_deleted', 'eq', False))
     for node in nodes:
-        node.update_search()
+        search.update_node(node, index=index)
         n_iter += 1
 
     logger.info('Nodes migrated: {}'.format(n_iter))
 
 
-def migrate_users():
+def migrate_users(index):
+    logger.info("Migrating users")
     n_migr = 0
     n_iter = 0
     for user in User.find():
         if user.is_active:
-            user.update_search()
+            search.update_user(user, index=index)
             n_migr += 1
         n_iter += 1
 
     logger.info('Users iterated: {0}\nUsers migrated: {1}'.format(n_iter, n_migr))
 
 
-def main():
+def migrate(delete, index=settings.ELASTIC_INDEX):
 
     script_utils.add_file_logger(logger, __file__)
     ctx = app.test_request_context()
     ctx.push()
+    new_index = set_up_index(index)
+
+    migrate_nodes(new_index)
+    migrate_users(new_index)
 
-    search.delete_all()
-    search.create_index()
-    migrate_nodes()
-    migrate_users()
+    set_up_alias(index, new_index)
+
+    if delete:
+        delete_old(new_index)
 
     ctx.pop()
 
 
+def set_up_index(idx):
+    alias = es.indices.get_aliases(index=idx)
+
+    if not alias or not alias.keys() or idx in alias.keys():
```

Comment:
Nitpicky, but the check for `not alias.keys()` is probably unnecessary. If `alias` is a `dict` with no keys, then `not alias` will be falsy.